pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -98,8 +98,15 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
     }
 
     // Check that table exists and is enabled
-    IdealState currentIdealState =
-        _helixDataAccessor.getProperty(_helixDataAccessor.keyBuilder().idealStates(tableNameWithType));
+    IdealState currentIdealState;
+    try {
+      currentIdealState =
+          _helixDataAccessor.getProperty(_helixDataAccessor.keyBuilder().idealStates(tableNameWithType));
+    } catch (Exception e) {
+      LOGGER.warn("Caught exception while fetching IdealState for table: {}", tableNameWithType, e);
+      return new RebalanceResult(RebalanceResult.Status.FAILED, "Caught exception while fetching IdealState: " + e,
```

Comment:
Not related to this PR but you can create a new constructor like `RebalanceResult(Status status, String description)`, so that you don't need to pass the null value all the time.

Reply:
I prefer making it more explicit haha